extensions/console/internal.m

```diff
@@ -0,0 +1,382 @@
+static int console_inputBackgroundColor(lua_State *L) {
+    NSTextField *input = [MJConsoleWindowController singleton].inputField ;
+
+    if (lua_type(L, 1) != LUA_TNONE) {
+        luaL_checktype(L, 1, LUA_TTABLE) ;
+        [input setBackgroundColor:[[LuaSkin shared] luaObjectAtIndex:1 toClass:"NSColor"]] ;
+    }
+
+    [[LuaSkin shared] pushNSObject:[input backgroundColor]] ;
+    return 1 ;
+}
+
+/// hs.console.smartInsertDeleteEnabled([flag]) -> bool
+/// Function
+/// Determine whether or not objects copied from the console window insert or delete space around selected words to preserve proper spacing and punctuation.
+///
+/// Parameters:
+///  * flag - an optional boolean value indicating whether or not "smart" space behavior is enabled when copying from the Hammerspoon console.
+///
+/// Returns:
+///  * the current value
+///
+/// Notes:
+///  * this only applies to future copy operations from the Hammerspoon console -- anything already in the clipboard is not affected.
+static int console_smartInsertDeleteEnabled(lua_State *L) {
+    NSTextView *output = [MJConsoleWindowController singleton].outputView ;
+
+    if (lua_type(L, 1) != LUA_TNONE) {
+        [output setSmartInsertDeleteEnabled:(BOOL)lua_toboolean(L, 1)] ;
+    }
+
+    lua_pushboolean(L, [output smartInsertDeleteEnabled]) ;
+    return 1 ;
+}
+
+/// hs.console.getHistory() -> array
+/// Function
+/// Get the Hammerspoon console history as an array.
+///
+/// Parameters:
+///  * None
+///
+/// Returns:
+///  * an array containing the history of commands entered into the Hammerspoon console.
+static int console_getHistory(__unused lua_State *L) {
+    [[LuaSkin shared] checkArgs:LS_TBREAK] ;
+    MJConsoleWindowController *console = [MJConsoleWindowController singleton] ;
+
+    [[LuaSkin shared] pushNSObject:[console history]] ;
+    return 1 ;
+}
+
+/// hs.console.setConsole([styledText]) -> none
+/// Function
+/// Clear the Hammerspoon console output window.
+///
+/// Parameters:
+///  * styledText - an optional `hs.styledtext` object containing the text you wish to replace the Hammerspoon console output with.  If you do not provide an argument, the console is cleared of all content.
+///
+/// Returns:
+///  * None
+///
+/// Notes:
+///  * You can specify the console content as a string or as an `hs.styledtext` object in either userdata or table format.
+static int console_setConsole(lua_State *L) {
+    [[LuaSkin shared] checkArgs:LS_TANY | LS_TOPTIONAL, LS_TBREAK] ;
+    MJConsoleWindowController *console = [MJConsoleWindowController singleton] ;
+
+    if (lua_gettop(L) == 0)
+        [[console.outputView textStorage] performSelectorOnMainThread:@selector(setAttributedString:)
+                                           withObject:[[NSMutableAttributedString alloc] init]
+                                        waitUntilDone:YES];
+    else {
```

Comment:
I'm generally not a fan of braceless conditional blocks, but mixing braced and braceless seems like a fairly bad idea to me :)